server/lib/fusor/password_filter.rb

```diff
@@ -0,0 +1,85 @@
+##
+# PasswordFilter
+# ================
+# Extracts passwords from deployment objects.
+# Provides functionality for filtering sensitive data (passwords) from text.
+
+class PasswordFilter
+  def self.password_cache
```

Comment:
@jmrodri I've been told QCI doesn't support multiple deployments occurring at the same time. State (storage of cached_passwords) is stored after the user gets to the final step of QCI and clicks the "Deploy" button. This was an attempt to save database queries and leave the method signature for Fusor.log identical.

The first version of this code queried the database for all Deployment objects every time a log message was written. This didn't seem ideal.

The second version required the user to supply a Deployment object to the logger if they wanted relevant passwords to be filtered. I spoke with @jwmatthews on this and he mentioned it would be ideal if the user doesn't have to remember to supply an extra argument with every log message.

Version you're seeing is the result of the problems with version 1 and 2, and hearing that QCI doesn't support multiple Deployments at the same time.

Since I defined instance-methods directly on the class, the cached passwords should persist across user logins, although I haven't tested with multiple users.